app/controllers/users_controller.rb

```diff
@@ -22,10 +30,51 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by(id: params[:id])
+  end
+
+  def update
+    @user = User.find_by params[:id]
+    if @user.update user_params
+      flash[:success] = t "flash.profile_updated"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    user = User.find_by id: params[:id]
+    if user&.destroy
+      flash[:success] = t "flash.delete_user"
+    else
+      flash[:danger] = t "flash.delete_fail"
+    end
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params
       .require(:user).permit(:name, :email, :password, :password_confirmation)
   end
+
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t "flash.please_login"
+    redirect_to login_url
+  end
+
+  def correct_user
+    @user = User.find_by(id: params[:id])
+    redirect_to(root_url) unless @user == current_user
```

Comment:
nên có thêm flash để họ biết lý do nha
```suggestion
    return if @user == current_user
    
    flash .... # Báo message ko có quyền truy cập
    redirect_to(root_url) 
```